migrations/blocked_tasks.js

```diff
@@ -0,0 +1,74 @@
+function taskIsBlocked(task) {
+    for(dep of task.depends_on) {
+        if (dep.unattainable) {
+            return true
+        }
+    }
+
+    return false
+}
+
+function unattainableResolved(task) {
+    for (dep of task.depends_on) {
+        if (dep.unattainable == null) {
+            return false
+        }
+    }
+    return true
+}
+
+finishedStatuses = ["success","failed"]
+millisecondsPerDay = 86400000
+delta = millisecondsPerDay * 1
+
+startingDate = ISODate("2016-01-01T00:00:00Z")
+while(startingDate < new Date()) {
+    loops = 0
+    tasksSize = 0
+    while (true) {
+        tasks = db.tasks.find({"create_time": {"$gte": startingDate, "$lte": new Date(startingDate.getTime() + delta)}, "status":"undispatched", "depends_on.status": {"$in": ["success", "failed", "", "*"]}, "depends_on": {"$elemMatch":{"unattainable": {"$exists": false}}}}, {"depends_on":1}).toArray()
+        printjson(loops++)
+        if (tasks.length == 0 || tasks.length == tasksSize) {
```

Comment:
`tasks.length` is the length of the tasks array from the current iteration, but `tasksSize` is the length of the tasks array from the previous iteration. If I'm understanding this correctly, these numbers are unlikely to match, because the former is tasks from a day later than the latter.

Reply:
I the case where I thought the numbers will match is when the last iteration of the inner while loop does no work.
For example, if we have a single task left and that task defines dependencies that do not exist in the database so depTask == null and we skip updating that task. In that case, the last iteration will have seen tasksSize == 1 and this iteration sees tasks.length ==1, so we can't do any more work and should stop spinning.